iceoryx_examples/icediscovery_in_c/iox_c_find_service.c

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_binding_c/runtime.h"
+//! [include service discovery]
+#include "iceoryx_binding_c/service_discovery.h"
+//! [include service discovery]
+#include "sleep_for.h"
+
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+#define SEARCH_RESULT_CAPACITY 10
+
+bool keepRunning = true;
+
+const char APP_NAME[] = "iox-c-find-service";
+
+static void sigHandler(int signalValue)
+{
+    // Ignore unused variable warning
+    (void)signalValue;
+    // caught SIGINT or SIGTERM, now exit gracefully
+    keepRunning = false;
+}
+
+//! [print function to be applied to search results]
+void printSearchResult(const iox_service_description_t service)
+{
+    printf(
+        "- Service: %s, Instance: %s, Event: %s\n", service.serviceString, service.instanceString, service.eventString);
+}
+//! [print function to be applied to search results]
+
+//! [search function for all front devices]
+void searchFrontDevices(const iox_service_description_t service, void* count)
+{
+    if (strncmp("FrontLeft", service.instanceString, IOX_CONFIG_SERVICE_STRING_SIZE) == 0
```

Comment:
If clang-tidy spits out a warning here, please ignore it, see this little thing for instance:
```c
 #include <stdio.h>
 #include <string.h>
 #define ID_NAME 6
 int main() {
     const char service[ID_NAME] = "12453";
     return strncmp("12", service, 15); // clang-tidy warning
     return strncmp("12", service, 15 - 0); // NO clang-tidy warning
     return strncmp("12", service, 15 - 1230); // NO clang-tidy warning
 }

```

It may makes sense to add `-bugprone-not-null-terminated-result` as exclusion to our root `.clang-tidy` file

Reply:
I added the exclusion